zephyr/test/vmh.c

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+
+#include <adsp_memory_regions.h>
+#include <sof/boot_test.h>
+#include <sof/lib/regions_mm.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/ztest.h>
+
+LOG_MODULE_DECLARE(sof_boot_test, CONFIG_SOF_LOG_LEVEL);
+
+static int vmh_test(void)
+{
+	struct vmh_heap *h = vmh_init_heap(NULL, MEM_REG_ATTR_CORE_HEAP, 0, false);
+
+	if (!h)
+		return -ENOMEM;
+
+	return vmh_free_heap(h);
+}
+
+ZTEST(sof_boot, vmh)
+{
+	int ret = vmh_test();
+
+	if (ret < 0) {
+		LOG_ERR("vmh init failed: %d", ret);
+		ztest_test_fail();
+	} else {
+		ztest_test_pass();
+	}
+}
```

Comment:
in fact, thinking about this now, this code would probably be common for all SOF boot tests, so we can have a macro to be called in this case something like `SOF_ZTEST_BOOT(vmh, vmh_test);` I can try to do this now or we can try to switch to one like this later when we get at least a second such boot test

Reply:
done